### DISCUSSION TITLE
Backup Schedule Server Time vs. Local Time v5.4

### DISCUSSION THREAD

**Discussion by mario-zelger** [ravendb/ravendb#18770] [Q&A] [answered]
We recently updated our Docker containers from version `5.4.116` to `5.4.119`. Now, we've noticed that the backups (Server Wide Backup Task) are no longer being executed in regards to the configured timezone in the container. They only seem to use `UTC` now.

The timezone in the container is configured via `/etc/localtime`. No `TZ` environment variable is set.

```bash
root@ravendb-0:/opt/RavenDB/Server# zdump /etc/localtime
/etc/localtime  Tue Jul  2 09:40:25 2024 CEST
root@ravendb-0:/opt/RavenDB/Server# date
Tue Jul  2 09:40:27 CEST 2024
```

We can see that difference in the estimated next run in the Studio:

![image](https://github.com/ravendb/ravendb/assets/15825982/f382176a-7407-4bd3-8e76-d525367334f1)

Until version `5.4.116` the backups were executed according to `CEST` and not `UTC`. We couldn't find any info regarding this in the [What's new](https://ravendb.net/docs/article-page/5.4/Csharp/start/whats-new) section between the mentioned versions. The problem seems to be the same with version `5.4.200`.

Was there a change on what timezone settings RavenDB uses for the backup schedule?

**Comment by grisha-kotler** [comment]
Can you share the folder name that was created?

**Reply by mario-zelger** [reply]
Yes, of course.

Here's the folder name.
![image](https://github.com/ravendb/ravendb/assets/15825982/06619619-3416-4eff-8595-ac71715ee00e)

Additionally, here's the file name of the full backup.
![image](https://github.com/ravendb/ravendb/assets/15825982/b2fee8ef-7e81-4084-8ac9-9a5ad96f0ebe)

**Reply by mario-zelger** [reply]
@grisha-kotler Do you have any updates on this?

**Reply by efratshenhar** [reply]
There was a regression in the latest version regarding the backup times.
You can follow 
https://issues.hibernatingrhinos.com/issue/RavenDB-21636/SlowTests.Issues.RavenDB20084.ShouldUpdateBackupInfoIfDatabaseUnloaded

**Reply by mario-zelger** [reply]
@efratshenhar We just updated to `5.4.201` and it seems the issue is not fixed yet. If I'm correct the fix should be part of the `5.4.201` release right? At least there was the note about it.

> [Backup] Fixed an issue with the backup running on UTC instead of server's local time

I'm seeing the exact same behavior as described here before.

**Comment by efratshenhar** [comment]
Could you please send us the RavenDB logs? You can download them from the RavenDB studio.

**Reply by mario-zelger** [reply]
👍 I just sent the logs to support@ravendb.net.

**Comment by mario-zelger** [comment, marked as answer]
Problem solved by configuring the `TZ` environment variable to the correct time zone.

Why didn't we do that before? When we started using RavenDB in containers (for Kubernetes) the `TZ` variable did not have any effect. It looked like there was no time zone data available (e.g. via `tzdata` package). Therefore, we configured the time zone via `/etc/localtime`. We did this by mounting a host volume from the path `/usr/share/zoneinfo/<time zone>` and mapped it to `/etc/localtime`. This worked at least until version `5.4.116`.

The latest RavenDB Docker images currently inherit from `mcr.microsoft.com/dotnet/runtime-deps:8.0-jammy`. The .NET 8 `runtime-deps` base image contains `tzdata` and therefore the necessary time zone data. (see [this commit](https://github.com/dotnet/dotnet-docker/pull/4847/files#diff-2ada8c561d88ccac9b016a71a971b4c04378cdda1c2ad491459b71f32eaf6965) where they added `tzdata`). I did not verify if something changed with the .NET 7.0 image as it is now EOL.

With `tzdata` available we can use the `TZ` variable again and RavenDB happily uses this information. As far as I can tell, any configuration in `/etc/localtime` has no effect anymore.

Thanks to the RavenDB team to look through the logs and configurations nevertheles!
